### PR TITLE
ci: migrate flux-update-schedule-check to gha

### PR DESCRIPTION
**Pull request by shaeed-noordeen**
**What problem does this PR solve?**:

**Which issue(s) does this PR fix?**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

File: .github/workflows/flux-update-scheduled-check.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+      - name: Setup asdf
+        uses: asdf-vm/actions/setup@v2
+        env:
+          GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+
+      - name: Run tests
+        env:
+          GH_TOKEN: ${{ secrets.MESOSPHERECI_USER_TOKEN }}
+        run: make KOMMANDER_REPO_PATH=${GITHUB_WORKSPACE}/kommander make flux-update
```
Comment (cbuto) [inline .github/workflows/flux-update-scheduled-check.yml:28]: is there an example run for this? 👀

File: .github/workflows/flux-update-scheduled-check.yml [hunk 2]
```diff
@@ -0,0 +1,28 @@
+name: Flux Update Schedule Check
+on:
+  schedule:
+    - cron:  '0 12 * * SUN'
+
+jobs:
+  flex-schedule-check:
```
Comment (cbuto) [inline .github/workflows/flux-update-scheduled-check.yml:7]:
```suggestion
  flux-schedule-check:
```

File: .github/workflows/flux-update-scheduled-check.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    name: Flux Update Schedule Check
+    runs-on:
+      - self-hosted
+      - small
+    steps:
```
Comment (cbuto) [inline .github/workflows/flux-update-scheduled-check.yml:12]: lets limit this to only run on `main`, once we create release branches we don't need/want this job running on those branches.

File: .github/workflows/flux-update-scheduled-check.yml (hunk 1 above)
Comment (shaeed-noordeen) [reply .github/workflows/flux-update-scheduled-check.yml:29]: https://github.com/mesosphere/kommander-applications/actions/runs/6299669767/job/17100858663

File: .github/workflows/flux-update-scheduled-check.yml [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+        with:
+          ssh-private-key: ${{ secrets.PRIVATE_SSH_KEY }}
+
+      - name: Setup asdf
+        uses: asdf-vm/actions/setup@v2
+        env:
+          GITHUB_API_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+
+      - name: Run tests
```
Comment (cbuto) [inline .github/workflows/flux-update-scheduled-check.yml:26]:
```suggestion
      - name: Check for Flux update
```

**Review by cbuto — APPROVED** [review]
just a small nit

File: .github/workflows/flux-update-scheduled-check.yml [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+  flux-schedule-check:
+    if: github.ref == 'refs/heads/main'
+    name: Flux Update Schedule Check
+    runs-on:
+      - self-hosted
+      - small
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v3
+      - uses: webfactory/ssh-agent@v0.7.0
```
Comment (cbuto) [inline .github/workflows/flux-update-scheduled-check.yml:16]: i think we can drop this

**Pull request merged**